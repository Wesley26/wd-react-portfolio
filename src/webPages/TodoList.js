import React, { useEffect, useState } from 'react';
import {
        toDoTitle,
        storageText,
        } from './contentMaster/PageContents.js';

import ToDoForm from './todoComponents/todoForm.js';
import TodoListing from './todoComponents/todoListing.js';

const TodoList = () => {

    const [todos, setTodos] = useState([]);
    const [todosMem, setTodosMem] = useState([]);

    useEffect(() => {

        localStorage.setItem('rememberTodo', todos);

        let memoryTodo = localStorage.getItem('rememberTodo');
        //console.log(memoryTodo); //uncomment to see localStorage in action

        if (memoryTodo) {
            setTodosMem(memoryTodo);
        };

    }, [todos]);

    console.log(todosMem); //uncomment to see localStorage in action

    return (
        
        <div className="font-body bg-body-lightGray m-6 shadow-xl">

            <div className="px-5 md:px-40 py-3">

                <h1 className="text-xl p-3 text-center">
                    {toDoTitle}
                </h1>

                <h3 className="text-md p-3 text-center">
                    {storageText}{localStorage.getItem('rememberTodo')}
                </h3>

                <ToDoForm 
                    saveToDo={todoText => {

                        if (todoText) {
                            /**
                             * Handle Todos from current window
                             */
                            const trimmedText = todoText.trim();

                            if (trimmedText.length > 0) {
                                setTodos([...todos, trimmedText]);
                            };
                        };
                    }}
                />

                <TodoListing 
                    todos={todos}
                    deleteTodo={todoIndex => {
                        const newTodos = todos.filter((_, index) => index !== todoIndex);

                        setTodos(newTodos);
                    }}
                />

            </div>
            
        </div>
    );
}

export default TodoList;