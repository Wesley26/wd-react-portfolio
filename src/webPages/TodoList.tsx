import { FC, useEffect, useState } from 'react';
import PageContents from './contentMaster/PageContents';

import ToDoForm from './todoComponents/todoForm';
import TodoListing from './todoComponents/todoListing';

const TodoList:FC = () => {

    const pageContents = PageContents();
    const [todos, setTodos] = useState<any>([]);
    const [todosMem, setTodosMem] = useState<any>([]);

    useEffect(() => {

        localStorage.setItem('rememberTodo', todos);

        let memoryTodo = localStorage.getItem('rememberTodo');
        //console.log(memoryTodo); //uncomment to see localStorage in action

        if (memoryTodo) {
            setTodosMem(memoryTodo);
        };

    }, [todos]);

    console.log(todosMem); //uncomment to see localStorage in action

    return (
        
        <div className="font-body bg-body_lightGray m-6 shadow-xl">

            <div className="px-5 md:px-40 py-3">

                <h1 className="text-xl p-3 text-center">
                    {pageContents.toDoTitle}
                </h1>

                <h3 className="text-md p-3 text-center">
                    {pageContents.storageText}{localStorage.getItem('rememberTodo')}
                </h3>

                <ToDoForm 
                    saveTodo={(todoText: any) => {

                        if (todoText) {
                            /**
                             * Handle Todos from current window
                             */
                            const trimmedText = todoText.trim();

                            if (trimmedText.length > 0) {
                                setTodos([...todos, trimmedText]);
                            };
                        };
                    }}
                />

                <TodoListing 
                    todos={todos}
                    deleteTodo={(todoIndex: any) => {
                        const newTodos = todos.filter((_: any, index: any) => index !== todoIndex);

                        setTodos(newTodos);
                    }}
                />

            </div>
            
        </div>
    );
}

export default TodoList;