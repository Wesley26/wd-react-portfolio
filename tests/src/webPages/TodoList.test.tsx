import { render } from '@testing-library/react';
import TodoList from '../../../src/webPages/TodoList';
import '../../src/index.css';

it(`render ContactMe.tsx test component`, () => {

    const view = (render(<TodoList />));

    expect(view).toMatchSnapshot();

});