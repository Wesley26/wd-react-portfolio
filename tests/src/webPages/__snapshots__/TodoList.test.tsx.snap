// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render ContactMe.tsx test component 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="font-body bg-body_lightGray m-6 shadow-xl"
      >
         
        <div
          class="px-5 md:px-40 py-3"
          style="opacity: 0; transform: translateX(0px) translateY(-20px) translateZ(0);"
        >
          <h1
            class="text-xl p-3 text-center"
          >
            My Development To-do List
          </h1>
          <h3
            class="text-md p-3 text-center"
          >
            Your current locally stored to-do list items: 
          </h3>
          <div>
            <form>
              <div
                class="MuiFormControl-root MuiFormControl-marginNormal MuiTextField-root css-1rv3ei8-MuiFormControl-root-MuiTextField-root"
              >
                <div
                  class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
                >
                  <input
                    aria-invalid="false"
                    class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                    id=":r0:"
                    placeholder="Add to-do item"
                    type="text"
                    value=""
                  />
                  <fieldset
                    aria-hidden="true"
                    class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                  >
                    <legend
                      class="css-ihdtdm"
                    >
                      <span
                        class="notranslate"
                      >
                        ​
                      </span>
                    </legend>
                  </fieldset>
                </div>
              </div>
            </form>
          </div>
          <ul
            class="MuiList-root MuiList-padding css-h4y409-MuiList-root"
          />
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="font-body bg-body_lightGray m-6 shadow-xl"
    >
       
      <div
        class="px-5 md:px-40 py-3"
        style="opacity: 0; transform: translateX(0px) translateY(-20px) translateZ(0);"
      >
        <h1
          class="text-xl p-3 text-center"
        >
          My Development To-do List
        </h1>
        <h3
          class="text-md p-3 text-center"
        >
          Your current locally stored to-do list items: 
        </h3>
        <div>
          <form>
            <div
              class="MuiFormControl-root MuiFormControl-marginNormal MuiTextField-root css-1rv3ei8-MuiFormControl-root-MuiTextField-root"
            >
              <div
                class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
              >
                <input
                  aria-invalid="false"
                  class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                  id=":r0:"
                  placeholder="Add to-do item"
                  type="text"
                  value=""
                />
                <fieldset
                  aria-hidden="true"
                  class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    class="css-ihdtdm"
                  >
                    <span
                      class="notranslate"
                    >
                      ​
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
          </form>
        </div>
        <ul
          class="MuiList-root MuiList-padding css-h4y409-MuiList-root"
        />
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
